bindings/python/py_engine.cpp

```diff
@@ -234,6 +235,32 @@ static PyObject* MaatEngine_load(PyObject* self, PyObject* args, PyObject* keywo
         }
     }
 
+    if (py_virtual_fs != nullptr)
+    {
+        // Check if it's a dict
+        if( !PyDict_Check(py_virtual_fs) )
+        {
+            return PyErr_Format(PyExc_TypeError, "'virtual_fs' parameter must be a dict");
+        }
+        PyObject *key, *value;
+        Py_ssize_t pos = 0;
+        while (PyDict_Next(py_virtual_fs, &pos, &key, &value))
+        {
+            if (not PyUnicode_Check(key))
```

Comment:
Python uses a [Easier to ask for forgiveness than permission](https://docs.python.org/3/glossary.html#term-EAFP) pattern usually and it seems you could probably directly call `PyUnicode_AsUTF8` instead of `PyUnicode_Check` first.

The code seems to agree on that - the `PyUnicode_AsUTF8` calls another function which does perform its own `PyUnicode_Check`, see: https://github.com/python/cpython/blob/7fc39a21cb85163a456eab91b52e5fe85e7f7e3e/Objects/unicodeobject.c#L4180-L4186

(well, maybe a question here is if `PyErr_BadArgument();` will not break anything on our side)